pandas/core/generic.py

```diff
@@ -7709,9 +7709,11 @@ def resample(
         convention: str = "start",
         kind: Optional[str] = None,
         loffset=None,
-        base: int = 0,
+        base: Optional[int] = None,
         on=None,
         level=None,
+        origin=None,
+        offset=None,
```

Comment:
```suggestion
        offset: Union[str, pd.Timedelta] = None,
```

I *think* this is what we need here; since you are wrapping this call in Timedelta might need to match whatever that constructor accepts, or explicitly require a Timedelta instance to be passed

Reply:
More on the same subject here: https://github.com/pandas-dev/pandas/pull/31809#discussion_r406554007